add: SIG-Scheduling annual report (#8288)
kubernetes/community: sig-scheduling/annual-report-2024.md

   - Governance and leadership changes
-->

- **The performance improvement**
  - Have put a lot of effort into QueueingHint ([KEP-4247](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4247-queueinghint)), which enhances the scheduler's retrying efficiency.
  - Introduced the asynchronous preemption ([KEP-4832](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4832-async-preemption)), which improves the scheduling throughput with the preemption.
  - Improved [the internal scheduling performance test (scheduler-perf)](https://github.com/kubernetes/kubernetes/tree/master/test/integration/scheduler_perf) to cover more scenarios and alert us when the degradation is introduced.
- **DRA**: Improved the DRA scheduling with the structured parameters ([KEP-4381](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4381-dra-structured-parameters#kube-scheduler)).

2. Are there any areas and/or subprojects that your group needs help with (e.g. fewer than 2 active OWNERS)?

> None.

Author: This is WIP. Need to check with the owners at each project.

Here's the checklist, I checked the ones I own plus I know there are enough maintainers.

- [ ]    cluster-capacity
- [ ]    descheduler 
- [x]    kube-scheduler-simulator
- [x]    kube-scheduler-wasm-extension
- [x]    kueue
- [ ]    kwok
- [x]    scheduler
- [ ]    scheduler-plugins

@a7i @ingvagabund @wzshiming @Huang-Wei Hi, I just randomly picked up one person per project. Can you help me answer this question, do you have enough maintainers at your project?

Author: /hold

Until we get the answers from all.

Reviewer: scheduler-plugins could really benefit from having another active maintainer.

Reviewer 2: cluster-capacity has only a single maintainer. Though, all the changes are related to k8s rebases only.
@a7i descheduler has ~3 active maintainers? We get help from community members from time to time.

Reviewer 3: KWOK doesn't have enough active maintainers, although not much has changed.

Author: Thanks all, added scheduler plugins and kwok here.

/unhold

3. Did you have community-wide updates in 2024 (e.g. KubeCon talks)?

<!--
  Examples include links to email, slides, or recordings.
-->

- [Kubecon EU 2024: SIG_Scheduling Intro & Deep Dive](https://sched.co/1YhjR)
- [Kubecon NA 2024: SIG_Scheduling Intro & Updates](https://sched.co/1hovV)

4. KEP work in 2024 (v1.30, v1.31, v1.32):

  - Alpha
    - [4816 - DRA Prioritized List](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4816-dra-prioritized-list) - v1.32
    - [4832 - Asynchronous Preemption](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4832-async-preemption) - v1.32

  - Beta
    - [3633 - Introduce MatchLabelKeys and MismatchLabelKeys to PodAffinity and PodAntiAffinity](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/3633-matchlabelkeys-to-podaffinity) - v1.31

>     - [4247 - Per-plugin callback functions for efficient requeueing in the scheduling queue](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/4247-queueinghint/README.md) - v1.32

Author: It was lacking. The metadata is fixed at https://github.com/kubernetes/enhancements/pull/5050.

   - Governance and leadership changes
-->

> - **The performance improvement**

Reviewer 4: ```suggestion
- **Performance improvements**
```

>   - Have put a lot of effort into QueueingHint ([KEP-4247](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4247-queueinghint)), which enhances the scheduler's retrying efficiency.

Reviewer 4: Maybe something like: Iterated on QueueingHints towards re-enabling the feature.

>   - Introduced the asynchronous preemption ([KEP-4832](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4832-async-preemption)), which improves the scheduling throughput with the preemption.

Reviewer 4: ```suggestion
  - Introduced asynchronous preemption ([KEP-4832](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/4832-async-preemption)), which improves the scheduling throughput when there are pods to be preempted.
```

>   - Improved [the internal scheduling performance test (scheduler-perf)](https://github.com/kubernetes/kubernetes/tree/master/test/integration/scheduler_perf) to cover more scenarios and alert us when the degradation is introduced.

Reviewer 4: ```suggestion
  - Improved [the internal scheduling performance test suite (scheduler-perf)](https://github.com/kubernetes/kubernetes/tree/master/test/integration/scheduler_perf) to cover more scenarios and alert us when the degradation is introduced.
```

> - **DRA**: Improved the DRA scheduling with the structured parameters ([KEP-4381](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4381-dra-structured-parameters#kube-scheduler)).

Reviewer 4: ```suggestion
- **DRA**: Added DRA scheduling with the structured parameters ([KEP-4381](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4381-dra-structured-parameters#kube-scheduler)).
```

Reviewer 5: Maybe add at the end ", replacing classic unstructured DRA."

## Operational

Operational tasks in [sig-governance.md]:
- [x] [README.md] reviewed for accuracy and updated if needed
- [x] [CONTRIBUTING.md] reviewed for accuracy and updated if needed

> - [ ] Other contributing docs (e.g. in devel dir or contributor guide) reviewed for accuracy and updated if needed

Author: So, actually I don't think we have paid attention enough to [these docs](https://github.com/kubernetes/community/tree/master/contributors/devel/sig-scheduling). We should review and update some docs (e.g., need to mention QHint at the queue doc etc)

Author: Opened https://github.com/kubernetes/community/issues/8289 to track this problem
